mitodl/micromasters
# Add inline-validation

#### What are the relevant tickets?

this closes #1510 

#### What's this PR do?

This implements inline-validation, so that all profile fields are validated in-line as the user fills them out, instead of waiting until the user presses save the first time to start doing validation.

#### How should this be manually tested?

OK it's a bit tricky. But basically, on `/profile` and `/learner` all profile fields (including on education and work history) should validate as-you-type, so that, for instance:

- if you tab to the first input (first name) and delete the input and tab away you should see a validation error
- if you click 'save' on `/profile` with a blank profile you should see all validation messages appear, which will disappear as you deal with them.
- in general, errors should disappear as you fix them
- fancier date valiation (like start date cannot be before end date) should happen in-line as well.

There are probably edge cases! Let's find them!

#### Any background context you want to provide?

#### Screenshots (if appropriate)

#### What GIF best describes this PR or how it makes you feel?
